### DISCUSSION TITLE
How to filter EmailMessage on the view

### DISCUSSION THREAD

**Discussion by mkv929** [deejay-hub/timeline-lwc#214] [Q&A] [answered]
I tried the configuration guide with the detailed steps under the section "**Filtering records**" but somehow it is not working for me.  My requirement is to filter out all the **emailmessages** from support portal team with xyz@email.com.  Is this possible to achieve?  How can this be done?

Note that - The view is on Account layout. So basically **Account** timeline shows tasks and events not directly linked to the Account but for all the indirect (child) relationships.

Thank you for your guidance.

Br,
Manoj

**Comment by deejay-hub** [comment]
Hi @mkv929 . For us to better understand the issue here it would be best for you to share your currently active custom metadata types for the Account object. Including the inclusion_field__c.
<img width="2106" alt="image" src="https://user-images.githubusercontent.com/7021883/192392025-4dc28ed7-0b3d-4632-ab8d-49e5a8c5ed7c.png">

Then share the formula you are using for the Inclusion Field. It isn't clear yet whether this is on the Activity Object or the EmailMessage Object.

**Reply by mkv929** [reply]
Hi @deejay-hub,

Thank you for your comment and looking into this.  Below is the screenshot of the active custom metadata types for the Account object. Including the inclusion_field__c.

![Account Metadata](https://user-images.githubusercontent.com/114398690/192501811-13bbc725-297b-417e-a51b-7087c47cc3f0.png)

**Formula that I'm using on Activity object**
![Activity Formula](https://user-images.githubusercontent.com/114398690/192501876-c402ebbe-83ba-40d1-aa85-a0a2cd460eb4.png)

However, I just noticed that the item showing on the timeline is NOT **emailmessage** and instead the _related task (grey coloured)_.  But I do want to filter OUT these as well. But somehow, not able to do so.  

![image](https://user-images.githubusercontent.com/114398690/192502787-aaca2390-2322-45c2-838d-0db7a87a0cb2.png)

Any guidance will really be appreciated.

Thank You,
Manoj

**Comment by deejay-hub** [comment, marked as answer]
Okay. So looks like you've asked to plot Open Activities and Activity History Only for the Account Object. Sadly Salesforce does not support applying WHERE clauses to these 2 relationship types so the inclusion field is ignored.

**Reply by mkv929** [reply]
Oh, that's too bad!   Is there any other way through which I can show all the activities related to account (directly/indirectly) with the WHERE clause?

Thank You,
Manoj

**Reply by deejay-hub** [reply]
Not as far as we are aware. Although all the source code is here so you're more than welcome to see what you can come up with. Given the restriction on WHERE clauses within an inner join you would have to apply post SOQL filtering after the record was returned OR query the objects separately. Both of these options are less than ideal from a performance standpoint. If you have any other suggestions...let us know.\

**Reply by mkv929** [reply]
Hi @deejay-hub , thanks for the feedback and guidance.  I'm not so good with SOQL but will try checking with dev team.  

Also, could you please help me to understand what metadata components should be used to show all the activities associated with the related objects of account (like contacts,Cases, etc..) on **Account layout**?

Thanks,
Manoj

**Reply by deejay-hub** [reply]
Time Warp plots **direct relationships** to objects where there is a lookup or master detail relationship. So you can plot cases, opportunities, custom objects etc where there is a lookup or master detail relating it to the parent Account. See plot standard, custom and junction objects in the config guide (https://salesforce.quip.com/ahFCA9VBEKtr).  So to see Cases on the Account then you just need to add the metadata record for that relationships.

Time Warp supports **indirect relationships** for the Account Object for Calls, Tasks, Events and Emails. This is configured as you have already done through OpenActivities and ActivityHistory relationships on the Account. Using this you'll see a task for a Contact related to the Account without the Task having a direct relationship to the Account.

If you want to see Cases directly related to the Parent Account then that can be done. If you want to see a Case related to a Contact (related to the Account) but not directly related with a lookup or master detail relationship then that isn't supported.

This is for performance reasons. Querying child to grandchild relationships would be expensive.

Hopefully that clarifies things.

_**Another (somewhat simplified) way to think about it. If you have a related list in the Object page then Time Warp can plot it. If you don't have a related list then it can't.**_

**Reply by mkv929** [reply]
Thank You @deejay-hub .  This is clear now.

**Reply by mkv929** [reply]
@deejay-hub Hi Dave, Does Time Warp follows org's security settings (OWD)?  OR it by passes?

I tried searching the details related to it but not able to find it so wanted to check with you.

Thank You,
Manoj

**Reply by deejay-hub** [reply]
@mkv929 yes OWD, sharing rules and field level security are followed.

**Reply by mkv929** [reply]
Thank You, @deejay-hub
